src/io/flutter/logging/FlutterLogView.java

```diff
@@ -45,20 +82,32 @@
     public SimpleTextAttributes style(@Nullable FlutterLogEntry entry, int attributes) {
       if (showColors && entry != null) {
         final FlutterLog.Level level = FlutterLog.Level.forValue(entry.getLevel());
-        // TODO(quangson91): use color from FlutterLogColorPage
         if (level != null) {
-          switch (level) {
-            case FINER:
-              return SimpleTextAttributes.GRAY_ATTRIBUTES;
-            case WARNING:
-              return ORANGE_ATTRIBUTES;
-            case SEVERE:
-              return SimpleTextAttributes.ERROR_ATTRIBUTES;
-          }
+          return getTextAttributesByLogLevel(level);
         }
       }
       return SimpleTextAttributes.REGULAR_ATTRIBUTES;
     }
+
+    @NotNull
+    private SimpleTextAttributes getTextAttributesByLogLevel(@NotNull Level level) {
+      if (textAttributesByLogLevelCached.containsKey(level)) {
+        return textAttributesByLogLevelCached.get(level);
+      }
+      try {
+        final TextAttributesKey key = LOG_LEVEL_TEXT_ATTRIBUTES_KEY_MAP.get(level);
+        final Color color = globalEditorColorsScheme.getAttributes(key).getForegroundColor();
+
+        final SimpleTextAttributes result = new SimpleTextAttributes(SimpleTextAttributes.STYLE_PLAIN, color);
+        textAttributesByLogLevelCached.put(level, result);
+        return result;
```

Comment:
I may be misreading, but does this ever get things from the cache?

Reply:
Also, how would a map of levels to colors get updated when `globalEditorColorsScheme` changes?

Another approach might be to keep a cache of colors to attributes...

For example:

```java
Map<Color, SimpleTextAttributes> attributeCache = new ConcurrentHashMap<>(); 

...

SimpleTextAttributes result = attributeCache.get(color); // or containsKey(...)
if (result == null) {
  result = ...
  attributeCache.put(color, result);
}
return result;
```

Then we don't have to worry about level cache consistency.

Reply:
I think we have to cache: `Map<Level, SimpleTextAttributes> textAttributesByLogLevelCache`.

Reason:
Cached values is used in method:
```java
public SimpleTextAttributes style(@Nullable FlutterLogEntry entry, int attributes) {
```

=> input
```java
final FlutterLog.Level level = FlutterLog.Level.forValue(entry.getLevel());
```
=> Ouput
```java
SimpleTextAttributes
```

=> `Map<Level, SimpleTextAttributes>` is right choise.

Reply:
When global color scheme changed -> we simply reset cache data.

https://github.com/flutter/flutter-intellij/pull/2399/commits/72ab8ff2668a61c0ff07747e2bca5dde782078ba#diff-2a35badbf1bca121cb7f36039ac67315R372

Reply:
> I may be misreading, but does this ever get things from the cache?

Yes, I tested. I'm sure it will get value from cache.